spec/models/supplier_location_spec.rb

```diff
@@ -43,4 +43,58 @@
       )
     end
   end
+
+  describe '.effective_on' do
+    subject(:effective_on) { described_class.effective_on(date) }
+
+    let(:location) { create(:location) }
+    let(:supplier) { create(:supplier) }
+    let(:date) { Date.today }
+
+    context 'with a nil date parameter' do
+      let!(:supplier_location1) { create(:supplier_location, supplier: supplier, location: location) }
+      let!(:supplier_location2) { create(:supplier_location, supplier: supplier, location: location, effective_from: Date.today, effective_to: Date.today) }
+      let(:date) { nil }
+
+      it 'returns matching supplier locations' do
```

Comment:
I did not fully understand this test:

with a nil date parameter
  returns matching supplier locations (only supplier_location1)

why? what's happen when date is nil?
it might be enough only to amend the test description, thanks